src/rendering/RenderUtil.cc

```diff
@@ -963,6 +981,31 @@ void RenderUtilPrivate::CreateRenderingEntities(
           return true;
         });
 
+    // particle emitters commands
+    _ecm.EachNew<components::ParticleEmitterCmd>(
+        [&](const Entity &_entityCmd,
+            const components::ParticleEmitterCmd *_emitterCmd) -> bool
+        {
+          // Find the particle emitter requested to be updated.
+          _ecm.Each<components::ParticleEmitter>(
+              [this, &_entityCmd, &_emitterCmd](const Entity &_entity,
+                  const components::ParticleEmitter *_emitter) -> bool
+              {
+                if (_emitterCmd->Data().name() == _emitter->Data().name())
+                {
+                  this->newParticleEmittersCmd.push_back(
+                    std::make_tuple(_entity, _emitterCmd->Data(), _entityCmd));
+                }
```

Comment:
```suggestion
                if (_emitterCmd->Data().name() == _emitter->Data().name())
                {
                  this->newParticleEmittersCmd.push_back(
                    std::make_tuple(_entity, _emitterCmd->Data(), _entityCmd));
                  return false;
                }
```
Once we have found the emitter that needs to be updated, we shouldn't need to continue searching for other emitters that match this emmiter command. We should just get the next emitter command message (if there are any more) and then run the search again.

Reply:
468b02ad76b3df9e24b862be4379d0ce25be5a3f